network.go

```diff
@@ -508,15 +508,29 @@ func (s *sandbox) removeNetwork() error {
 	}
 	defer netHandle.Delete()
 
+	// Create a duplicate of the route list, with no pointers,
+	// otherwise this leads to a mix when routes are removed.
+	var routeList []pb.Route
 	for _, route := range s.network.routes {
-		if err := s.removeRoute(netHandle, route); err != nil {
+		routeList = append(routeList, *route)
```

Comment:
@sboeuf Where else in the code are we updating this list? 
Why not just store objects instead of pointers in the list? We can always modify the objects itself when a slice is passed to a function.

Reply:
This involves a bigger change regarding how things are handled if we move from pointer to object. I am fine with any solution but I want some feedback before to go ahead and implement it.
@bergwolf @jodh-intel @egernst @grahamwhaley PTAL

Reply:
This is because a slice is not really a good data structure for dynamical element removal, - you cannot avoid data copy if you want to remove within a loop.

OTOH, if we add an ID to pb.Route to uniquely identify a route for a sandbox, we can use a map for it and it is perfectly safe to do
```
for _, r := range routes {
     delete(routes, r.ID)
}
```

The same applies to pb.Interface where we can use interface.Name to uniquely identify an interface and thus use a map instead of a slice to store it.